contributing/code/standards.rst

```diff
@@ -265,19 +252,28 @@ Service Naming Conventions
 Documentation
 ~~~~~~~~~~~~~
 
-* Add PHPDoc blocks for all classes, methods, and functions (though you may
-  be asked to remove PHPDoc that do not add value);
+* Add PHPDoc blocks for classes, methods, and functions only when they add
+  relevant information that does not duplicate the name, native type
+  declaration or context (e.g. ``instanceof`` checks);
+
+* Only use annotations and types defined in `the PHPdoc reference`_. In
+  order to improve types for static analysis, the following annotations are
+  also allowed:
+
+  * `Generics`_, with the exception of ``@template-covariant``.
```

Comment:
Why do we make that exception?

Reply:
As far as I know, this is not yet supported by PHPstan.

Reply:
phpstan supports it (otherwise, they would not have [this blog post](https://phpstan.org/blog/whats-up-with-template-covariant) explaining it)